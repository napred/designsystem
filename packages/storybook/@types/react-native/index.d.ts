// hack because react-native conflicts with @types/node and lib.dom
declare module 'react-native' {
  import { ComponentType } from 'react';

  type Style = {
    default: any;
    create(obj: any): any;
  };

  const StyleSheet: Style;
  const Image: ComponentType;
  const Text: ComponentType;
  const View: ComponentType;

  export { Image, StyleSheet, Text, View };

  export type RegisteredStyle<T> = any;
}
