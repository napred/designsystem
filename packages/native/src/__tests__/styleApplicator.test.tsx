import React from 'react';
import { Text, View } from 'react-native';
import { render } from 'react-native-testing-library';
import { createComponent, createSimpleCache, DesignSystem } from '../';

jest.mock('Dimensions');

const DSText = createComponent('DSText', Text);
const DSView = createComponent('DSView', View);

const BoxDistinct = createComponent('BoxDistinct', View);
BoxDistinct.defaultProps = { 
  bgColor: 'primary',
  color: 'greyLightest',
 };

const BoxDistinct2 = createComponent('BoxDistinct2', Text);
const BoxDistinct3 = createComponent('BoxDistinct3', Text);

DSText.defaultProps = {
  color: 'green',
};

DSView.defaultProps = {
  bgColor: 'green',
  color: 'white',
};

describe('style applicator', () => {
  it('works correctly for simple system component', () => {
    const { toJSON } = render(
      <DesignSystem>
        <DSView color="red" />
        <DSText color="blue" />
      </DesignSystem>,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('works correctly for extended system component', () => {
    const DSExtendedText = createComponent('DSExtendedText', DSText);
    const DSExtendedView = createComponent('DSExtendedView', DSView);

    const { toJSON } = render(
      <DesignSystem>
        <DSExtendedView bgColor="blue" />
        <DSExtendedText bgColor="tomato" m={[1, 2, 3, 4]} />
      </DesignSystem>,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('creates distinct cache for same props but distinct components', () => {
    const cache = createSimpleCache();
    const { toJSON } = render(
      <DesignSystem cache={cache}>
        <BoxDistinct />
        <BoxDistinct />
        <BoxDistinct2 />
        <BoxDistinct2 />
        <BoxDistinct3 />
      </DesignSystem>,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('works correctly for overridden system component', () => {
    const DSExtendedView = createComponent('DSExtendedView', DSView);

    DSExtendedView.defaultProps = {
      fontWeight: 'bold',
    };

    const { toJSON } = render(
      <DesignSystem>
        <DSView as={DSExtendedView} bgColor="blue" />
      </DesignSystem>,
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
