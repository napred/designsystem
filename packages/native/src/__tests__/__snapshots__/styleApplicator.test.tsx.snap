// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`style applicator works correctly for extended system component 1`] = `
Array [
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#3B5998",
          "color": "#FFFFFF",
          "fontFamily": "Palanquin",
          "fontSize": 16,
          "lineHeight": 24,
        },
      ]
    }
  />,
  <Text
    style={
      Array [
        Object {
          "backgroundColor": "tomato",
          "color": "#3CD048",
          "fontFamily": "Palanquin",
          "fontSize": 16,
          "lineHeight": 24,
          "marginBottom": 4,
          "marginLeft": 4,
          "marginRight": 4,
          "marginTop": 4,
        },
      ]
    }
  />,
]
`;

exports[`style applicator works correctly for overridden system component 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#3B5998",
        "color": "#FFFFFF",
        "fontFamily": "Palanquin",
        "fontSize": 16,
        "fontWeight": "bold",
        "lineHeight": 24,
      },
    ]
  }
/>
`;

exports[`style applicator works correctly for simple system component 1`] = `
Array [
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#3CD048",
          "color": "#EA2E49",
          "fontFamily": "Palanquin",
          "fontSize": 16,
          "lineHeight": 24,
        },
      ]
    }
  />,
  <Text
    style={
      Array [
        Object {
          "color": "#3B5998",
          "fontFamily": "Palanquin",
          "fontSize": 16,
          "lineHeight": 24,
        },
      ]
    }
  />,
]
`;
